packages/ui-components/src/components/absolute-time-picker-dropdown-input/absolute-time-picker-dropdown-input.tsx

```diff
@@ -109,20 +109,39 @@ export class KvAbsoluteTimePickerDropdownInput implements IAbsoluteTimePickerDro
 		}
 	};
 
+	private handleEmptyTextChange = () => {
+		if (this.mode === EAbsoluteTimePickerMode.Single) {
+			this.singleTimeInputValue = '';
+		} else {
+			if (this.focusedInput === EInputSource.From) {
+				this.rangeInputValues = { from: '', to: this.rangeInputValues.to };
+			} else {
+				this.rangeInputValues = { to: '', from: this.rangeInputValues.from };
+			}
+		}
+	};
+
 	private hanleOnTextChange = (event: CustomEvent<string>) => {
 		const parsedDateTime = dayjs(event.detail, DATE_FORMAT);
 
+		if (isEmpty(event.detail)) {
+			this.handleEmptyTextChange();
+			return;
+		}
+
 		if (parsedDateTime.isValid()) {
 			if (this.mode === EAbsoluteTimePickerMode.Single) {
 				this.singleTimeInputValue = parsedDateTime.format(DATE_FORMAT);
+				this.displayedMonth = parsedDateTime;
 				return;
 			}
 
 			if (this.focusedInput === EInputSource.From) {
-				this.rangeInputValues = { from: parsedDateTime.format(DATE_FORMAT), ...this.rangeInputValues };
+				this.rangeInputValues = { from: parsedDateTime.format(DATE_FORMAT), to: this.rangeInputValues.to };
```

Comment:
This change here and bellow are necessary for the correct behaviour of the calendar. If I use destructuring here, the component will ignore the new from I am providing. This way I can force the update of the object correctly.

Reply:
Previously, the destructuring wasn't in the wrong order?

```
this.rangeInputValues = { ...this.rangeInputValues, from: parsedDateTime.format(DATE_FORMAT)  };
```